lib/sync/default-dataHandlers.js

```diff
@@ -0,0 +1,79 @@
+var syncUtil = require('../sync-util');
+var util = require('util');
+
+var COLL_POSTFIX = '_collision';
+var mongo;
+
+module.exports = function (db) {
+
+  if (!db) {
+    throw new Error('MongoDB instance must be passed to module.');
+  }
+  mongo = db
+
+  return {
+    doList: function (dataset_id, params, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doList : ' + dataset_id +
+                     ' :: query_params=' + util.inspect(params) +
+                     ' :: meta_data=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).find(params).toArray(cb);
+    },
+
+    doCreate: function (dataset_id, data, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doCreate : ' + dataset_id + ' :: ' + util.inspect(data) +
+                     ' :: meta=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).insertOne(data, cb);
```

Comment:
I believe the `create` implementation have to at least return the `uid` (the `_id`) field as part of the callback.  See https://github.com/feedhenry/fh-mbaas-api/blob/master/lib/sync-datahandler.js#L47